### ISSUE TITLE
Bug Report about: Report a bug encountered while operating Ianvs labels

### ISSUE THREAD

**Issue by vishalmore90** [kubeedge/ianvs#616]
**What happened**:
There is a severe architectural violation in the `core` framework module. Specifically, the Test-Time Augmentation (TTA) and Active Boost single-task learning paradigms tightly couple the `core` framework to example-specific applications, relying on external examples and undeclared third-party dependencies (`torch`, `mmdet`). 

Because the `examples` directory lacks an `__init__.py` and is explicitly excluded during packaging by `find_packages()` in `setup.py`, any user who installs `ianvs` and instantiates these paradigms will encounter a `ModuleNotFoundError` for `examples`, `torch`, or `mmdet`.

Affected Files and Lines:
- `core/testcasecontroller/algorithm/paradigm/singletask_learning/singletask_learning_tta.py` (Lines 11-14)
- `core/testcasecontroller/algorithm/paradigm/singletask_learning/singletask_learning_active_boost.py` (Lines 13-15)

Root Cause:
- **Tight Coupling to Examples**: 
  - `singletask_learning_tta.py` directly imports `from examples.yaoba.singletask_learning_yolox_tta.resource.utils.TTA_strategy import TTA_Strategy`.
  - `singletask_learning_active_boost.py` directly imports `infer_anno`, `merge_predict_results`, `compute_error`, etc., from `examples.yaoba.singletask_learning_boost.resource.utils...`.
- **Missing Dependencies**:
  - `singletask_learning_tta.py` directly imports `torch` and `from mmdet.apis import init_detector`. These heavyweight ML libraries are not listed in the project's global `requirements.txt` or `setup.py`.

**What you expected to happen**:
1. The `core` framework should be entirely abstract, independent, and agnostic of specific example implementations. Example-specific logic (like `TTA_Strategy` or `infer_anno`) should reside completely within the `examples/` directory and should be passed/registered into the core framework via abstract interfaces.
2. The core framework should only rely on dependencies declared in the project's global `requirements.txt` or `setup.py`. Example-specific dependencies (like `torch` and `mmdet`) should not be directly imported inside the `core/` module.

**How to reproduce it (as minimally and precisely as possible)**:
1. Install the `ianvs` package (e.g., via `pip install .`) or rely on the core library without the `examples` folder present in your `PYTHONPATH`.
2. Attempt to run or import the SingleTaskLearningTTA or SingleTaskLearningACBoost paradigms:
   ```python
   from core.testcasecontroller.algorithm.paradigm.singletask_learning.singletask_learning_tta import SingleTaskLearningTTA
   ```
3. A `ModuleNotFoundError` is raised for `examples`, `torch`, or `mmdet`.

**Anything else we need to know?**:
Suggested Fix: Refactor `SingleTaskLearningTTA` and `SingleTaskLearningACBoost` to utilize abstract interfaces or callbacks for custom inference/augmentation strategies. Move the concrete implementations of `TTA_Strategy`, `infer_anno`, `merge_predict_results`, etc., back to the `examples/yaoba/` domain where they belong, and remove the `import examples...`, `import torch`, and `import mmdet` statements from the core library.

Environment:
- Ianvs Version: v0.3.0 / Latest main
- Python Version: 3.6+
